app/models/user.rb

```diff
@@ -1,11 +1,73 @@
 class User < ActiveRecord::Base
   has_many :rounds
   has_many :player_rounds
-  has_many :rounds, through: :player_rounds
   has_many :courses
   has_many :tees
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  def not_scored_rounds
+    not_scored = []
+      rounds.each do |round|
+        if round.round_date < Date.today
+          if round.score == nil
+            not_scored << round
+          end
+        end
+      end
+    not_scored.last(3)
+  end
+
+  def upcoming_rounds
+    upcoming = []
+    rounds.last(3).each do |round|
+      if round.round_date > Date.today
+        upcoming << round
+      end
+    end
+    upcoming
+  end
+
+  def handicap
```

Comment:
Assignment Branch Condition size for handicap is too high. [49.81/15]<br>Perceived complexity for handicap is too high. [20/7]